builtin/credential/approle/path_tidy_user_id.go

```diff
@@ -165,21 +165,27 @@ func (b *backend) tidySecretID(ctx context.Context, req *logical.Request) (*logi
 			// Accessor indexes were not getting cleaned up until 0.9.3. This is a fix
 			// to clean up the dangling accessor entries.
 			if len(accessorMap) > 0 {
+				logger.Trace(fmt.Sprintf("dangling accessorMap length: %d", len(accessorMap)))
+				// Wait to get a lock for each of the secretIDs to make sure we avoid a race conditon
+				// where the accessor has been written but not yet the secret
 				for _, lock := range b.secretIDLocks {
 					lock.Lock()
-					defer lock.Unlock()
+					lock.Unlock()
 				}
-				for accessorHash, _ := range accessorMap {
-					logger.Trace("found dangling accessor, verifying")
-					// Ideally, locking on accessors should be performed here too
-					// but for that, accessors are required in plaintext, which are
-					// not available. The code above helps but it may still be
-					// racy.
```

Comment:
I'm not certain that we should change this code. It sounds like it's already racy. I'm not clear on where we might place the unlock for the mutex, but it seems like anything that holds the lock for a shorter amount of time would result in _racier_ behavior. Perhaps I'm not thinking of it the right way and it'll be clearer once it's pushed.